src/configure/configurers/provisioningConfigurer.ts

```diff
@@ -0,0 +1,280 @@
+    private filesToCommit: DraftFile[] = [];
+
+    constructor(localGitRepoHelper: LocalGitRepoHelper){
+        this.localGitRepoHelper = localGitRepoHelper;
+    }
+
+    public async queueProvisioningPipelineJob(provisioningConfiguration: ProvisioningConfiguration, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+        try {
+            this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+            const OrgAndRepoDetails = wizardInputs.sourceRepository.repositoryId.split('/');
+            return await this.provisioningServiceClient.createProvisioningConfiguration(provisioningConfiguration, OrgAndRepoDetails[0], OrgAndRepoDetails[1]);
+        } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnableToCreateProvisioningPipeline, error);
+            throw error;
+        }
+    }
+
+    public async getProvisioningPipeline(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+       try {
+        this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+        return await this.provisioningServiceClient.getProvisioningConfiguration(jobId, githubOrg, repository);
+       } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnabletoGetProvisioningPipeline, error);
+            throw error;
+       }
+    }
+
+    public async awaitProvisioningPipelineJob(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration> {
+        let statusNotFound: number = 0;
+        try {
+            const provisioningServiceResponse = await this.getProvisioningPipeline(jobId, githubOrg, repository, wizardInputs);
+            if ( provisioningServiceResponse.result){
+                if ( provisioningServiceResponse.result.status ===  "Queued" ||  provisioningServiceResponse.result.status == "InProgress") {
+                await sleepForMilliSeconds(this.refreshTime);
+                return await this.awaitProvisioningPipelineJob(jobId, githubOrg, repository, wizardInputs);
+                } else if (provisioningServiceResponse.result.status ===  "Failed") {
+                throw new Error(provisioningServiceResponse.result.message) ;
+                } else {
+                    return provisioningServiceResponse;
+                }
+            }else {
+                if ( statusNotFound < this.maxNonStatusRetry ) {
+                    statusNotFound++;
+                    await sleepForMilliSeconds(this.refreshTime);
+                    return await this.awaitProvisioningPipelineJob(jobId, githubOrg, repository, wizardInputs);
+                } else {
+                 throw new Error("Failed to receive queued pipeline provisioning job status");
```

Comment:
we should throw the error returned by the API. Please do so other places as well.

Reply:
This is also for same case. Return response is 2XX , but result not received in response.

Reply:
ok